src/qml/filters/gpstext/ui.qml

```diff
@@ -0,0 +1,815 @@
+/*
+ * Copyright (c) 2014-2021 Meltytech, LLC
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+import QtQuick.Dialogs 1.2
+import QtQuick.Window 2.1
+import Shotcut.Controls 1.0 as Shotcut
+import org.shotcut.qml 1.0 as Shotcut
+import QtQml.Models 2.12
+
+Item {
+	id: gpsTextRoot
+	width: 300
+    height: 800
+	
+	property url settingsOpenPath: 'file:///' + settings.openPath
+	Shotcut.File { id: gpsFile }
+	
+	// This signal is used to workaround context properties not available in
+	// the FileDialog onAccepted signal handler on Qt 5.5.
```

Comment:
I think we do not care about Qt 5.5 since Qt 5.12 is the oldest that is supported.

Reply:
This is how it was in lut3d filter where I copied it from, I'm still not sure I understand what it's doing, Is it saving the last opened folder location?

Reply:
OK. You can probably just delete the comment. The comment is about the signal.

You are asking about the following line which saves the opened filed name in the Shotcut settings. It does this so that the next time the users tries to open a file Shotcut will start the file dialog in the same directory.

Reply:
Ok, I tried moving stuff around but had to revert to original version and just removed the comment now.